File: sample/src/main/java/contacts/sample/cheatsheet/blockednumbers/java/DeleteBlockedNumbersActivity.java
package contacts.sample.cheatsheet.blockednumbers.java;

import static contacts.core.WhereKt.contains;

import android.app.Activity;

import contacts.core.BlockedNumbersFields;
import contacts.core.ContactsFactory;
import contacts.core.blockednumbers.BlockedNumbersDelete;
import contacts.core.entities.BlockedNumber;

public class DeleteBlockedNumbersActivity extends Activity {

    BlockedNumbersDelete.Result deleteBlockedNumber(BlockedNumber blockedNumber) {
        return ContactsFactory.create(this)
                .blockedNumbers()
                .delete()
                .blockedNumbers(blockedNumber)
                .commit();
    }

    BlockedNumbersDelete.Result deleteBlockedNumberWithId(Long blockedNumberId) {
        return ContactsFactory.create(this)
                .blockedNumbers()
                .delete()
                .blockedNumbersWithId(blockedNumberId)
                .commit();
    }

    BlockedNumbersDelete.Result deleteBlockedNumbersContaining555() {
        return ContactsFactory.create(this)
                .blockedNumbers()
                .delete()
                .blockedNumbersWhere(contains(BlockedNumbersFields.Number, "555"))
                .commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/blockednumbers/kotlin/DeleteBlockedNumbersActivity.kt
package contacts.sample.cheatsheet.blockednumbers.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.blockednumbers.BlockedNumbersDelete
import contacts.core.contains
import contacts.core.entities.BlockedNumber

class DeleteBlockedNumbersActivity : Activity() {

    fun deleteBlockedNumber(blockedNumber: BlockedNumber): BlockedNumbersDelete.Result =
        Contacts(this)
            .blockedNumbers()
            .delete()
            .blockedNumbers(blockedNumber)
            .commit()

    fun deleteBlockedNumberWithId(blockedNumberId: Long): BlockedNumbersDelete.Result =
        Contacts(this)
            .blockedNumbers()
            .delete()
            .blockedNumbersWithId(blockedNumberId)
            .commit()

    fun deleteBlockedNumbersContaining555(): BlockedNumbersDelete.Result =
        Contacts(this)
            .blockedNumbers()
            .delete()
            .blockedNumbersWhere { Number contains "555" }
            .commit()
}